src/routes/graph/update-pin-groups.ts

```diff
@@ -0,0 +1,35 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+
+interface Request {
+  body: {
+    hashIds: string[];
+    mapLayer?: {
+      key: string,
+      includeEdges?: boolean,
+    };
+  };
+}
+
+type Response = void;
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'post',
+  path: '/update-pin-groups/:hashId',
```

Comment:
* Should the `:hashId` be dropped in the path?
* We never use the `update`-verb (or `add` or `delete`) in the path, as it is already in the method.

Shall we therefore change the path to `/pin-groups`?

Reply:
> We never use the update-verb (or add or delete) in the path, as it is already in the method.

I'd agree with removing it, the HTTP method should be an indicator of the operation already. That being said, now that I'm saying this, I notice that it's a `POST` and not a `PUT` 😅. But it's a problem that we already have in the codebase. And making a breaking change in the platform for this is very likely the last thing that we want to do, so, not a problem for me here.

> Should the :hashId be dropped in the path?

Nice catch, it's unused in the platform, and doesn't make a lot of sense in this case.

Reply:
Updated

Reply:
👍 

> Actually I think it should be a PATCH (partial update), but that's not the point

I agree. That's a mistake we made since the beginning, but consistently :).